benchmark/benchmark.cpp

```diff
@@ -18,69 +20,65 @@
 
 #define UMF_BENCHMARK_REGISTER_F(BaseClass, Method)                            \
     BENCHMARK_REGISTER_F(BaseClass, Method)                                    \
-        ->ArgNames(                                                            \
-            BENCHMARK_PRIVATE_CONCAT_NAME(BaseClass, Method)::argsName())      \
-        ->Name(BENCHMARK_PRIVATE_CONCAT_NAME(BaseClass, Method)::name())       \
-        ->Iterations(                                                          \
-            BENCHMARK_PRIVATE_CONCAT_NAME(BaseClass, Method)::iterations())
-
-UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, glibc_fix, fixed_alloc_size,
-                              glibc_malloc);
+        ->Apply(                                                               \
+            &BENCHMARK_PRIVATE_CONCAT_NAME(BaseClass, Method)::defaultArgs)
 
 // Benchmarks scenarios:
 
 // The benchmark arguments specified in Args() are, in order:
 // benchmark arguments, allocator arguments, size generator arguments.
 // The exact meaning of each argument depends on the benchmark, allocator, and size components used.
 // Refer to the 'argsName()' function in each component to find detailed descriptions of these arguments.
+
+static void default_alloc_fix_size(benchmark::internal::Benchmark *benchmark) {
+    benchmark->Args({10000, 0, 4096});
+    benchmark->Args({10000, 100000, 4096});
+    benchmark->Threads(4);
+    benchmark->Threads(1);
+}
+
+static void
+default_alloc_uniform_size(benchmark::internal::Benchmark *benchmark) {
+    benchmark->Args({10000, 0, 8, 64 * 1024, 8});
+    benchmark->Threads(4);
+    benchmark->Threads(1);
+}
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, glibc_fix, fixed_alloc_size,
+                              glibc_malloc);
+
 UMF_BENCHMARK_REGISTER_F(alloc_benchmark, glibc_fix)
-    ->Args({10000, 0, 4096})
-    ->Args({10000, 100000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_alloc_fix_size);
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, glibc_uniform,
                               uniform_alloc_size, glibc_malloc);
 UMF_BENCHMARK_REGISTER_F(alloc_benchmark, glibc_uniform)
-    ->Args({10000, 0, 8, 64 * 1024, 8})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_alloc_uniform_size);
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, os_provider, fixed_alloc_size,
                               provider_allocator<os_provider>);
 UMF_BENCHMARK_REGISTER_F(alloc_benchmark, os_provider)
-    ->Args({10000, 0, 4096})
-    ->Args({10000, 100000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_alloc_fix_size);
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, proxy_pool, fixed_alloc_size,
                               pool_allocator<proxy_pool<os_provider>>);
 
 UMF_BENCHMARK_REGISTER_F(alloc_benchmark, proxy_pool)
-    ->Args({1000, 0, 4096})
-    ->Args({1000, 100000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_alloc_fix_size);
```

Comment:
soo this one had 1k instead of 10k iterations? perhaps was that on purpose before? if not, pls disregard

Reply:
This is exactly why i did this patch - so eliminate copypaste errors - now is correct